### PR TITLE
[JBJCA-1354] Remove invocation to connection listener from inside the…

### PR DESCRIPTION
**Pull request by maeste**
… synchronized flush block, thus resolving the potential for deadlock situations

### CODE REVIEW

File: core/src/main/java/org/jboss/jca/core/connectionmanager/pool/mcp/SemaphoreArrayListManagedConnectionPool.java [hunk 1]
```diff
@@ -698,9 +701,9 @@ public void removeIdleConnections()
       {
          for (ConnectionListener cl : destroy)
          {
-            log.tracef("Destroying timedout connection %s", cl);
```
Comment (fl4via) [inline core/src/main/java/org/jboss/jca/core/connectionmanager/pool/mcp/SemaphoreArrayListManagedConnectionPool.java]: I'm not sure why the log message is different here... maybe the other branch had a different message, or did I add this difference in the log messages without realizing it, in the original commit?

File: core/src/main/java/org/jboss/jca/core/connectionmanager/pool/mcp/SemaphoreArrayListManagedConnectionPool.java [hunk 2]
```diff
@@ -876,11 +892,9 @@ private ConnectionListener createConnectionEventListener(Subject subject, Connec
    }
 
    /**
-    * Destroy a connection
-    *
-    * @param cl the connection to destroy
+    * {@inheritDoc}
     */
-   void doDestroy(ConnectionListener cl)
+   public void connectionListenerDestroyed(ConnectionListener cl)
```
Comment (fl4via) [inline core/src/main/java/org/jboss/jca/core/connectionmanager/pool/mcp/SemaphoreArrayListManagedConnectionPool.java]: Only the statistics update should remain in the connectionListenerDestroyed method. The remainder is the part that we moved to inside ConnectionListener.destroy.

File: core/src/main/java/org/jboss/jca/core/connectionmanager/pool/mcp/SemaphoreConcurrentLinkedQueueManagedConnectionPool.java [hunk 3]
```diff
@@ -1126,6 +1129,35 @@ private void removeConnectionListenerFromPool(ConnectionListenerWrapper clw)
...
+   public void connectionListenerDestroyed(ConnectionListener cl)
+   {
+      if (cl.getState() == ConnectionState.DESTROYED)
+      {
+         log.tracef("ManagedConnection is already destroyed %s", cl);
+
+         return;
+      }
+
+      if (statistics.isEnabled())
```
Comment (fl4via) [inline core/src/main/java/org/jboss/jca/core/connectionmanager/pool/mcp/SemaphoreConcurrentLinkedQueueManagedConnectionPool.java]: The same here. Only this statistics update should remain inside connectionListenerDestroyed. The remainder of the code is already in ConnectionListener.destroy

**Comment by maeste** [comment]
Apllied all Flavia's requested changes

**Pull request merged**